utils/dev-tool/src/main.rs

```diff
@@ -0,0 +1,102 @@
+use std::path::PathBuf;
+use structopt::StructOpt;
+
+pub mod android;
+pub mod apple;
+pub mod server;
+pub mod setup;
+pub mod utils;
+pub mod workspace;
+
+#[derive(Debug, PartialEq, StructOpt)]
+#[structopt(about = "Lockbook's development and ci tool.")]
+enum Commands {
+    /// Verify CI environment
+    VerifyCIEnvironment,
+
+    /// Check the formatting
+    FmtCheck,
+
+    /// Check the lint
+    ClippyCheck,
+
+    /// Check the formatting of the android client
+    AndroidFmtCheck,
+
+    /// Check the lint of the android client
+    AndroidLintCheck,
+
+    /// Make kotlin jni libs
+    MakeKotlinLibs,
+
+    /// Make kotlin jni libs for tests
+    MakeKotlinTestLib,
+
+    /// Make swift jni libs for tests
+    MakeSwiftTestLib,
+
+    /// Build server
+    BuildServer,
+
+    /// Run server detached
+    RunServer,
+
+    /// Run all rust tests
+    RunRustTests,
+
+    /// Run kotlin integration tests
+    RunKotlinTests,
+
+    /// Run swift integration tests
+    RunSwiftTests,
+
+    /// Kill server for commit hash
+    KillServer,
+
+    /// Kill all servers running
+    KillAllServers,
+}
+
+#[derive(Clone)]
+pub struct ToolEnvironment {
+    root_dir: PathBuf,
+    target_dir: PathBuf,
+    hash_info_dir: PathBuf,
+    commit_hash: String,
+}
+
+impl Default for ToolEnvironment {
+    fn default() -> Self {
+        let (root_dir, target_dir, hash_info_dir) = utils::get_dirs();
```

Comment:
you should either move this impl into utils and inline the logic here, or you should have 3 fns that get you these three values